rpc/rpc.h

```diff
@@ -243,10 +247,23 @@ namespace rpc
     };
 
     extern "C" Stub* new_rpc_stub(IStream* stream, bool ownership = false);
-    extern "C" StubPool* new_stub_pool(uint64_t expiration, uint64_t connect_timeout, uint64_t rpc_timeout);
-    extern "C" StubPool* new_uds_stub_pool(const char* path, uint64_t expiration,
-                                uint64_t connect_timeout,
-                                uint64_t rpc_timeout);
+    /**
+    About timeout:
+    1. When a socket/stub not used by any caller for `expiration` microsecs, it will be dropped.
+    2. When socket connecting, it will fail by be timed out after `connect_timeout` microsecs.
+    3. When socket needs to read/write (like sending request and waiting for any response), but got
+       nothing in `transfer_timeout` microsecs, it will be timed out. It's just a timeout for a single
+       read/write operation, not a RPC timeout, which is controlled by the caller when invoking `Stub::call`.
+    4. `Stub::call` measures the time from invoking `call` before sending request to received
+       response head. Receiving response body is not considered.
+    **/
+    extern "C" StubPool* new_stub_pool(uint64_t expiration,
+                                       uint64_t connect_timeout,
+                                       uint64_t transfer_timeout);
```

Comment:
rpc_timeout?

Reply:
it's actually transfer timeout

Reply:
> transfer_timeout -> rpc_timeout?

When socket needs to read/write (like sending request and waiting for any response), but got nothing in `transfer_timeout` microsecs, it will be timed out. It's just a timeout for a single read/write operation, not a RPC timeout, which is controlled by the caller when invoking `Stub::call`.

That means that if on same stream, a heartbeat RPC keep beeping, the stream keeps reading and writing without trigger this timeout. Other concurrent RPC call on this stream will not report as timedout by `transfer_timeout`. So I think `rpc_timeout` is not proper name for it.